### PR TITLE
use architect orb

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Hey guys I need help here. I am not sure how the Circle Config for operators should look like. Couldn't find anything in the `architect-orb` docs. Uncertain which project may already does something similar. This here is pretty much the first operator I move to the new orb.

**Comment by rossf7** [comment]
@xh3b4sd I think `build` should be `architect/build` and that will fix the error.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,58 +1,20 @@
...
   build:
     jobs:
+      - architect/go-test:
+          name: go-test
+          filters:
+            # Trigger job also on git tag.
+            tags:
+              only: /^v.*/
+
       - build
```
Comment (xh3b4sd) [inline .circleci/config.yml]:
```suggestion
      - architect/build
```

**Comment by xh3b4sd** [comment]
Doesn't seem to help. Can I check somewhere where this is already done? Also, when we have the build step, do we need the testing step? Do we need the build step when we have the testing step? I lack a bit of insights here and would appreciate talking with someone who knows more. 🙈

**Comment by rossf7** [comment]
> Can I check somewhere where this is already done?

Mr @kopiczko likely knows better but I think you're blazing a trail. I don't know of any operators that are not using the build step. 

My knowledge here is limited but I think the build step is needed. For example for push-to-docker so there is a binary. There should also be some build cache magic going on. Sorry I can't be more help.

**Comment by xh3b4sd** [comment]
Discussed with Pawel. The operator will be flattened and Team Magic takes care of the Circle Config later. Discarding this here now.

**Pull request closed without merging**